scripts/image_app.py

```diff
@@ -0,0 +1,45 @@
+import streamlit as st
+from openai import OpenAI
+from prediction_market_agent_tooling.markets.markets import (
+    MarketType,
+    get_binary_markets,
+)
+
+st.title("Prediction market thumbnail generator")
+
+markets = get_binary_markets(42, MarketType.OMEN)
+
+custom_question_input = st.checkbox("Provide a custom question", value=False)
+question = (
+    st.text_input("Question")
+    if custom_question_input
+    else st.selectbox("Select a question", [m.question for m in markets])
+)
+
+prompt_template = st.text_area(
+    "Prompt",
+    value="""Create a thumbnail image for the following prediction market '{market}'.
+Don't write any text on the image.""",
+)
+
+market = (
+    [m for m in markets if m.question == question][0]
+    if not custom_question_input
+    # If custom question is provided, just take some random market and update its question.
+    else markets[0].model_copy(update={"question": question, "current_p_yes": 0.5})
+)
+
+with st.spinner("Generating image..."):
+    response = OpenAI().images.generate(
+        model="dall-e-3",
+        prompt=prompt_template.format(market=market.question),
+        size="1024x1024",
+        quality="standard",
+        n=1,
+    )
```

Comment:
Since you just need the question as prompt and the market query is only needed for when there is no input, I would suggest
-> Create a DataFrame with the first 5 markets (sorted by CLOSING SOONEST, for ex) and offer this as default input, for people to get inspired - people can copy-paste one of the market titles into the input field

Reply:
I don't understand. Why they can't just select the market from the dropdown?

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/03da8f76-8d48-4152-9d20-c9536c0b1b35)

Reply:
That works for me, thanks for the screenshot.